consensus/src/pipeline/virtual_processor/processor.rs

```diff
@@ -304,20 +308,41 @@ impl VirtualStateProcessor {
     }
 
     fn maybe_update_pruning_point_and_candidate(self: &Arc<Self>) {
-        // TODO: call this method from resolve_virtual
+        let virtual_sp = self.virtual_state_store.read().get().unwrap().ghostdag_data.selected_parent;
+        if virtual_sp == self.genesis_hash {
+            return;
+        }
+
+        let ghostdag_data = self.ghostdag_store.get_compact_data(virtual_sp).unwrap();
         let pruning_read_guard = self.pruning_store.upgradable_read();
         let current_pp = pruning_read_guard.pruning_point().unwrap();
```

Comment:
Change to `pruning_read_guard.get()` to obtain the full `PruningPointInfo` and use index and candidate below (in both if/else branches) instead of re-accessing the cache